subsys/bluetooth/audio/bap_broadcast_sink.c

```diff
@@ -457,13 +462,21 @@ static bool net_buf_decode_subgroup(struct net_buf_simple *buf,
 	 * broadcasted BASEs
 	 */
 	while (ltv_buf.len != 0) {
-		struct bt_codec_data *codec_data = &codec->data[codec->data_count++];
+		struct bt_codec_data *codec_data;
+
+		if (codec->data_count >= ARRAY_SIZE(codec->data)) {
+			LOG_WRN("BIS codec data overflow; discarding");
+			break;
+		}
```

Comment:
Two comments here:
1) This does not need to be check for each loop iteration, right? The values are constant, and this check can be moved to before the loop.
2) I'm not sure whether we should `break` or `return false` here. This happens when we cannot store all the data. In the Unicast Client (`unicast_client_ep_set_codec`) we actually either store all, or none, so perhaps we should discard the entire BASE here as well if we cannot fully decode it. 

Both of these comments also apply to the metadata handling below

Reply:
1. No: `codec->data_count` is incremented in each iteration of the loop.

Reply:
> Ideally we'd only keep the ones we actually care about and discard the rest.

That makes sense, but on the other hand, it may also give the user of the Broadcast Sink implementation only part of the information in the BASE, a (proper non-log) way to indicate that the BASE information has been truncated. 

I don't think there is a specific right or wrong answer here, and I'm unsure which I think is best :)

Reply:
> 1. No: `codec->data_count` is incremented in each iteration of the loop.

You are correct :)

Reply:
I believe this truncation issue links back to https://github.com/zephyrproject-rtos/zephyr/issues/42860
If the received TLV data was pushed to the application it could decide which values to store on the stream, which to store locally, and which to ignore. 
Some profiles require certain data(codec data/meta data) - hence which TLV values to store on the stream could be based on which profiles was enabled in the build - and the amount of storage could match that. But we still need to notify the application about the remaining "unused" values - as the application might need them for something. So maybe the solution could be:
1. In the stack only store TLV data that the enabled profiles needs.
2. Rather than discarding the data introduce an optional API function call-back to deliver what is discarded (maybe make the pointer to the value a const and state that the lifetime of the data is only for the duration of the call-back - to avoid memory allocation).
But that should be another PR I think.